src/components/annotations/draw.js

```diff
@@ -487,15 +487,15 @@ function drawRaw(gd, options, index, subplotId, xa, ya) {
             // to get the parity of the number of intersections.
             if(edges.reduce(function(a, x) {
                 return a ^
-                    !!lineIntersect(headX, headY, headX + 1e6, headY + 1e6,
+                    !!Lib.segmentsIntersect(headX, headY, headX + 1e6, headY + 1e6,
```

Comment:
Unless that equates to an `all` condition where you actually need to check all. Not 100% sure. The `^` is a bit confusing.